### ISSUE TITLE
(lsp-runtime-lib) Compile error without `USE_LIBSNDFILE`: ordered comparison of pointer with integer zero

### ISSUE THREAD

**Issue by Hanziness** [sadko4u/lsp-plugins#297]
Hello!

I have recently attempted to set up a build workflow for Windows (using MSYS2) and came across a compile error that seems not so Windows-specific, though it compiles fine on Linux. The error is coming from [`lsp-runtime-lib/main/mm/OutAudioFileStream.cpp` line 358](https://github.com/lsp-plugins/lsp-runtime-lib/blob/fd6e87e292f9e4ef4273d74c053484c174f1f660/src/main/mm/OutAudioFileStream.cpp#L358), where `g++` fails with the following message:

```
   g++  [lsp-runtime-lib] main/mm/OutAudioFileStream.cpp
mkdir -p /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-runtime-lib/main/mm/
g++ -o /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-runtime-lib/main/mm/OutAudioFileStream.o -c main/mm/OutAudioFileStream.cpp -fPIC -march=x86-64 -m64 -O2 -fvisibility=hidden -fno-exceptions -fno-rtti -fdata-sections -ffunction-sections -fno-asynchronous-unwind-tables -pipe -Wall -DLSP_RUNTIME_LIB_BUILTIN -fvisibility=hidden -DUSE_LSP_COMMON_LIB -DUSE_LSP_LLTL_LIB -DUSE_LIBSHLWAPI -DUSE_LIBWINMM -DUSE_LIBMSACM -DUSE_LIBMPR   -I"/d/a/lsp-plugins/lsp-plugins/modules/lsp-common-lib/include" -DLSP_COMMON_LIB_BUILTIN -I"/d/a/lsp-plugins/lsp-plugins/modules/lsp-lltl-lib/include" -DLSP_LLTL_LIB_BUILTIN     -I"/d/a/lsp-plugins/lsp-plugins/modules/lsp-runtime-lib/include" -DLSP_RUNTIME_LIB_BUILTIN
main/mm/OutAudioFileStream.cpp: In member function 'virtual lsp::status_t lsp::mm::OutAudioFileStream::open(const lsp::LSPString*, const lsp::mm::audio_stream_t*, size_t)':
main/mm/OutAudioFileStream.cpp:358:21: error: ordered comparison of pointer with integer zero ('const lsp::mm::audio_stream_t*' and 'int')
  358 |             if (fmt <= 0)
      |                 ~~~~^~~~
make[5]: *** [Makefile:131: /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-runtime-lib/main/mm/OutAudioFileStream.o] Error 1
```

From what I could find, [marking this as an error is the intended behavior of the compiler](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87699), so it really seems like an error. With a quick glance at the code, I believe that this originates from [line 252](https://github.com/lsp-plugins/lsp-runtime-lib/blob/fd6e87e292f9e4ef4273d74c053484c174f1f660/src/main/mm/OutAudioFileStream.cpp#L252), where the failing code is not included if `USE_LIBSNDFILE` is set (which is probably not on Windows).

The full build log is [here](https://github.com/Hanziness/lsp-plugins/actions/runs/4036424277/jobs/6939009670).

Oh, additionally, I have `libsndfile` installed in the build environment, so there could some other reason while the build does not pick it up. I'll try to look into it in the meantime.

**Comment by sadko4u** [comment]
It seems that you're doing something strange while trying to build the source code for Windows.

To build the code, you just need the MinGW compiler and the GNU Make tools that come with it.

It seems that your build tries to access the `libsndfile` library but for Windows, the `lsp-runtime-lib` uses the `winmm` and `msacm` libraries to implement the same functions.

So the main case is that the build system probably doesn't see your environment as `Windows`.

**Comment by sadko4u** [comment]
Oh, sorry. Now I see that it really sees the `winmm` dependency. Seems to be some error that my `MinGW` compiler didn't notice.

**Comment by sadko4u** [comment]
Oh, it's just typo. The `sfmt` variable should be used instead of the `fmt` argument.

Here's the change just committed into the `devel` branch
https://github.com/lsp-plugins/lsp-runtime-lib/commit/d0ec947e807a713956f59f307f56626b7e630e86

**Comment by sadko4u** [comment]
Please let me know here if you have additional compilation problems.

**Comment by Hanziness** [comment]
Amazing, thank you very much for the quick response! In the meantime I'll experiment with building on Windows by getting `libsndfile` recognized during build. I believe we can close this issue, I'll get back with a Windows-specific one if I find something :)

**Comment by sadko4u** [comment]
The issue will be closed after the 1.2.6 release will be issued.

**Comment by sadko4u** [comment]
Builds in 1.2.6. Not tested yet for proper work under Windows
